.github/workflows/draft-release.yml

```diff
@@ -61,85 +72,61 @@ jobs:
       - name: Store release tarball as artifact
         uses: actions/upload-artifact@v2
         with:
-          name: tarball
+          name: release-tarball
           path: ${{ env.TARBALL_PATH }}
           if-no-files-found: error
 
+  create-packages:
+    needs: create-release-tarball
+    uses: ./.github/workflows/create-packages.yml
+    with:
+      source-tarball-artifact-name: release-tarball
+      dbld-image-mode: build
+
+  upload-packages:
+    needs: create-packages
+    uses: ./.github/workflows/upload-packages.yml
+    with:
+      pkg-type: stable
+    secrets:
+      azure-sas-token: ${{ secrets.AZURE_SAS_TOKEN }}
+
+  create-draft-release:
+    runs-on: ubuntu-latest
+    needs: [create-release-tarball, upload-packages]
+    steps:
+      - name: Checkout syslog-ng source
+        uses: actions/checkout@v2
+
+      - name: Download release tarball artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: release-tarball
+
       - name: Create draft release
         run: |
-          sed "1 i${RELEASE_NAME}\n" NEWS.md > /tmp/message
+          sed "1 i${{ needs.create-release-tarball.outputs.RELEASE_NAME }}\n" NEWS.md > /tmp/message
 
           hub release create \
             --draft \
             --file /tmp/message \
-            --attach ${TARBALL_PATH} \
-            ${RELEASE_TAG}
+            --attach ${{ needs.create-release-tarball.outputs.TARBALL_NAME }} \
+            ${{ needs.create-release-tarball.outputs.RELEASE_TAG }}
 
+  comment-workflow-result:
+    runs-on: ubuntu-latest
+    needs: create-draft-release
+    if: always() && github.event_name == 'pull_request' && needs.create-draft-release.result != 'skipped'
```

Comment:
Note to self:

Maybe instead of `needs.create-draft-release.result != 'skipped'`, `needs.create-release-tarball != 'skipped'` would be better. I think the `create-draft-release` job can be skipped, if a previous job failed, and in this case we want to comment the failure.

Reply:
Good idea, thanks.

I used `${{ success() }}` and `${{ cancelled() }}` in the commenting instead.

I prefer the `always()` + `pull_request` contidions, because `always()` runs in case of `cancelled()`, too. I also added a comment for the `cancelled()` result.

Reply:
`${{ success() }}` and `${{ cancelled() }}` does not work in the script :( I falled back to `result`...

Reply:
I had to remove the cancelled comment, as I cannot check the whole workflow's status, I would need to check each previous checks if they were cancelled manually, and write the necessary comment.

This comment just causes so much trouble....

Reply:
I gave it another try and I think your latest patch is perfect for the job (or workflow... pun intended).

You are using the expression `if: (github.event.pull_request.merged && contains(github.event.pull_request.labels.*.name, 'version-bump'))` in the `comment-workflow-result` job, which is the same as used in initial job `create-release-tarball`.

Since the `draft-release` runs on every closed PR (either merged or not), we need some kind of filtering when to run the draft-release workflow. We had the very same filter before you refactored `draft-release`, and we need it for every job in the workflow.
I've checked github actions doc for an alternative, but haven't found a better way.